[Bug]: Wrong status code

### Brief problem description

assert 404 == 200
E        +  where 404 = <Response [404 Not Found]>.status_code
Status code is 404.

### Steps to Reproduce

uv run poe test

### Expected Result

Status code 200 expected

### Actual Result

Status code 404

--- a/src/app/services/item_service.py
+++ b/src/app/services/item_service.py
@@ -228,7 +228,7 @@ def get_item_by_id_dfs_iterative(
 
                     for task in lab.tasks:
                         counter += 1
-                        if lab.id == item_id:
+                        if task.id == item_id:
                             return FoundItem(task, counter)
 
                         for step in task.steps:

--- a/src/app/services/item_service.py
+++ b/src/app/services/item_service.py
@@ -228,6 +228,7 @@ def get_item_by_id_dfs_iterative(
 
                     for task in lab.tasks:
                         counter += 1
+
                         if task.id == item_id:
                             return FoundItem(task, counter)
 

